Task_1_5_1/src/main/java/Operation.java

```diff
@@ -0,0 +1,115 @@
+import java.util.EmptyStackException;
+import java.util.Stack;
+
+import static java.lang.Double.NaN;
+
+/**
+ * Класс операций.
+ */
+public class Operation {
+    String name;
+    int argumentCount;
+
+    /**
+     * Конструктор класса.
+     *
+     * @param operation - строковое представление операции.
+     * @throws IllegalArgumentException - если операция невозможна.
+     */
+    public Operation(String operation) throws IllegalArgumentException {
+        name = operation;
+        switch (name) {
+            case "+":
+                argumentCount = 2;
+                break;
+            case "-":
+                argumentCount = 2;
+                break;
+            case "*":
+                argumentCount = 2;
+                break;
+            case "/":
+                argumentCount = 2;
+                break;
+            case "pow":
+                argumentCount = 2;
+                break;
+            case "log":
+                argumentCount = 1;
+                break;
+            case "sin":
+                argumentCount = 1;
+                break;
+            case "cos":
+                argumentCount = 1;
+                break;
+            case "sqrt":
+                argumentCount = 1;
+                break;
+            default:
+                throw new IllegalArgumentException();
+        };
```

Comment:
В современных версиях JAVA это можно сделать гораздо более лаконично
```suggestion
        argumentCount = switch (name) {
            case "+", "-", "*", "/", "pow" ->  2;
            case "log", "sin", "cos", "sqrt" ->  1;
            default -> throw new IllegalArgumentException();
        };
```

Reply:
Здраствуйте, изначально так и было сделано, но в таком случае gradlew build не проходит (см. commit https://github.com/mixdone/OOP/pull/8/commits/fcc8845d8fe71f735b77895507d2d14a5b93c22d)

Reply:
Аа, понятно... Тогда пусть остаётся так. 
Видимо старая версия java на CI, и я пока не могу понять, где она задаётся. Посмотрю ещё попозже